app-libs/stf/poc/match-test/src/lib.rs

```diff
@@ -1,15 +1,23 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+
 use serde::{Deserialize, Serialize};
 
+#[cfg(not(feature = "std"))]
+use alloc::string::String;
```

Comment:
Some things, like most heap allocated stuff, is not by default available in `no_std`, and you need to import it from the `alloc` crate. Note:

```rust
#[cfg(not(feature = "std"))]
extern crate alloc;
```
Is only needed at the crate root, and not in submodules of a crate.